File: backend/tests/fixtures/taint/keys.go
package fixtures

import (
	"crypto/rand"
	"crypto/rsa"
)

// legacyBits is a package-level const. OpenGrep does NOT resolve these in Go,
// so the wrapper below is reached through a direct literal instead. See
// test_taint_scanner.py::test_module_level_constants_are_a_known_boundary.
const legacyBits = 1024

func buildKey(bits int) (*rsa.PrivateKey, error) {
	return rsa.GenerateKey(rand.Reader, bits)
}

// LegacyKey passes a literal through the wrapper — this is resolved.
func LegacyKey() (*rsa.PrivateKey, error) {
	return buildKey(1024)
}

// ConstKey passes a package const — this is NOT resolved.
func ConstKey() (*rsa.PrivateKey, error) {
	return buildKey(legacyBits)
}

File: backend/tests/fixtures/taint/keys.rb
require 'openssl'

# A top-level constant is not resolved through a wrapper in Ruby, so the
# resolved case below passes a literal. See
# test_taint_scanner.py::test_module_level_constants_are_a_known_boundary.
LEGACY_BITS = 1024

def build_key(bits)
  OpenSSL::PKey::RSA.new(bits)
end

def legacy_key
  build_key(1024)
end

def const_key
  build_key(LEGACY_BITS)
end

def build_cipher(name)
  OpenSSL::Cipher.new(name)
end

def legacy_cipher
  build_cipher('DES-EDE3-CBC')
end
